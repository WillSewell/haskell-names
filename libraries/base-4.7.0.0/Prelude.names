[{"name":"Bool","entity":"data","module":"GHC.Types"},{"name":"False","entity":"constructor","type":"Bool","module":"GHC.Types"},{"name":"True","entity":"constructor","type":"Bool","module":"GHC.Types"},{"name":"&&","entity":"value","module":"GHC.Classes"},{"name":"||","entity":"value","module":"GHC.Classes"},{"name":"not","entity":"value","module":"GHC.Classes"},{"name":"otherwise","entity":"value","module":"GHC.Base"},{"name":"Maybe","entity":"data","module":"Data.Maybe"},{"name":"Nothing","entity":"constructor","type":"Maybe","module":"Data.Maybe"},{"name":"Just","entity":"constructor","type":"Maybe","module":"Data.Maybe"},{"name":"maybe","entity":"value","module":"Data.Maybe"},{"name":"Either","entity":"data","module":"Data.Either"},{"name":"Left","entity":"constructor","type":"Either","module":"Data.Either"},{"name":"Right","entity":"constructor","type":"Either","module":"Data.Either"},{"name":"either","entity":"value","module":"Data.Either"},{"name":"Ordering","entity":"data","module":"GHC.Types"},{"name":"LT","entity":"constructor","type":"Ordering","module":"GHC.Types"},{"name":"EQ","entity":"constructor","type":"Ordering","module":"GHC.Types"},{"name":"GT","entity":"constructor","type":"Ordering","module":"GHC.Types"},{"name":"Char","entity":"data","module":"GHC.Types"},{"name":"String","entity":"type","module":"GHC.Base"},{"name":"fst","entity":"value","module":"Data.Tuple"},{"name":"snd","entity":"value","module":"Data.Tuple"},{"name":"curry","entity":"value","module":"Data.Tuple"},{"name":"uncurry","entity":"value","module":"Data.Tuple"},{"name":"Eq","entity":"class","module":"GHC.Classes"},{"name":"==","class":"Eq","entity":"method","module":"GHC.Classes"},{"name":"/=","class":"Eq","entity":"method","module":"GHC.Classes"},{"name":"Ord","entity":"class","module":"GHC.Classes"},{"name":"compare","class":"Ord","entity":"method","module":"GHC.Classes"},{"name":"\u003c","class":"Ord","entity":"method","module":"GHC.Classes"},{"name":"\u003c=","class":"Ord","entity":"method","module":"GHC.Classes"},{"name":"\u003e=","class":"Ord","entity":"method","module":"GHC.Classes"},{"name":"\u003e","class":"Ord","entity":"method","module":"GHC.Classes"},{"name":"max","class":"Ord","entity":"method","module":"GHC.Classes"},{"name":"min","class":"Ord","entity":"method","module":"GHC.Classes"},{"name":"Enum","entity":"class","module":"GHC.Enum"},{"name":"succ","class":"Enum","entity":"method","module":"GHC.Enum"},{"name":"pred","class":"Enum","entity":"method","module":"GHC.Enum"},{"name":"toEnum","class":"Enum","entity":"method","module":"GHC.Enum"},{"name":"fromEnum","class":"Enum","entity":"method","module":"GHC.Enum"},{"name":"enumFrom","class":"Enum","entity":"method","module":"GHC.Enum"},{"name":"enumFromThen","class":"Enum","entity":"method","module":"GHC.Enum"},{"name":"enumFromTo","class":"Enum","entity":"method","module":"GHC.Enum"},{"name":"enumFromThenTo","class":"Enum","entity":"method","module":"GHC.Enum"},{"name":"Bounded","entity":"class","module":"GHC.Enum"},{"name":"minBound","class":"Bounded","entity":"method","module":"GHC.Enum"},{"name":"maxBound","class":"Bounded","entity":"method","module":"GHC.Enum"},{"name":"Int","entity":"data","module":"GHC.Types"},{"name":"Integer","entity":"data","module":"GHC.Integer.Type"},{"name":"Float","entity":"data","module":"GHC.Types"},{"name":"Double","entity":"data","module":"GHC.Types"},{"name":"Rational","entity":"type","module":"GHC.Real"},{"name":"Num","entity":"class","module":"GHC.Num"},{"name":"+","class":"Num","entity":"method","module":"GHC.Num"},{"name":"-","class":"Num","entity":"method","module":"GHC.Num"},{"name":"*","class":"Num","entity":"method","module":"GHC.Num"},{"name":"negate","class":"Num","entity":"method","module":"GHC.Num"},{"name":"abs","class":"Num","entity":"method","module":"GHC.Num"},{"name":"signum","class":"Num","entity":"method","module":"GHC.Num"},{"name":"fromInteger","class":"Num","entity":"method","module":"GHC.Num"},{"name":"Real","entity":"class","module":"GHC.Real"},{"name":"toRational","class":"Real","entity":"method","module":"GHC.Real"},{"name":"Integral","entity":"class","module":"GHC.Real"},{"name":"quot","class":"Integral","entity":"method","module":"GHC.Real"},{"name":"rem","class":"Integral","entity":"method","module":"GHC.Real"},{"name":"div","class":"Integral","entity":"method","module":"GHC.Real"},{"name":"mod","class":"Integral","entity":"method","module":"GHC.Real"},{"name":"quotRem","class":"Integral","entity":"method","module":"GHC.Real"},{"name":"divMod","class":"Integral","entity":"method","module":"GHC.Real"},{"name":"toInteger","class":"Integral","entity":"method","module":"GHC.Real"},{"name":"Fractional","entity":"class","module":"GHC.Real"},{"name":"/","class":"Fractional","entity":"method","module":"GHC.Real"},{"name":"recip","class":"Fractional","entity":"method","module":"GHC.Real"},{"name":"fromRational","class":"Fractional","entity":"method","module":"GHC.Real"},{"name":"Floating","entity":"class","module":"GHC.Float"},{"name":"pi","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"exp","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"log","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"sqrt","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"**","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"logBase","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"sin","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"cos","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"tan","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"asin","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"acos","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"atan","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"sinh","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"cosh","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"tanh","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"asinh","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"acosh","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"atanh","class":"Floating","entity":"method","module":"GHC.Float"},{"name":"RealFrac","entity":"class","module":"GHC.Real"},{"name":"properFraction","class":"RealFrac","entity":"method","module":"GHC.Real"},{"name":"truncate","class":"RealFrac","entity":"method","module":"GHC.Real"},{"name":"round","class":"RealFrac","entity":"method","module":"GHC.Real"},{"name":"ceiling","class":"RealFrac","entity":"method","module":"GHC.Real"},{"name":"floor","class":"RealFrac","entity":"method","module":"GHC.Real"},{"name":"RealFloat","entity":"class","module":"GHC.Float"},{"name":"floatRadix","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"floatDigits","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"floatRange","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"decodeFloat","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"encodeFloat","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"exponent","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"significand","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"scaleFloat","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"isNaN","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"isInfinite","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"isDenormalized","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"isIEEE","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"isNegativeZero","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"atan2","class":"RealFloat","entity":"method","module":"GHC.Float"},{"name":"subtract","entity":"value","module":"GHC.Num"},{"name":"even","entity":"value","module":"GHC.Real"},{"name":"odd","entity":"value","module":"GHC.Real"},{"name":"gcd","entity":"value","module":"GHC.Real"},{"name":"lcm","entity":"value","module":"GHC.Real"},{"name":"^","entity":"value","module":"GHC.Real"},{"name":"^^","entity":"value","module":"GHC.Real"},{"name":"fromIntegral","entity":"value","module":"GHC.Real"},{"name":"realToFrac","entity":"value","module":"GHC.Real"},{"name":"Monad","entity":"class","module":"GHC.Base"},{"name":"\u003e\u003e=","class":"Monad","entity":"method","module":"GHC.Base"},{"name":"\u003e\u003e","class":"Monad","entity":"method","module":"GHC.Base"},{"name":"return","class":"Monad","entity":"method","module":"GHC.Base"},{"name":"fail","class":"Monad","entity":"method","module":"GHC.Base"},{"name":"Functor","entity":"class","module":"GHC.Base"},{"name":"fmap","class":"Functor","entity":"method","module":"GHC.Base"},{"name":"mapM","entity":"value","module":"Control.Monad"},{"name":"mapM_","entity":"value","module":"Control.Monad"},{"name":"sequence","entity":"value","module":"Control.Monad"},{"name":"sequence_","entity":"value","module":"Control.Monad"},{"name":"=\u003c\u003c","entity":"value","module":"Control.Monad"},{"name":"id","entity":"value","module":"GHC.Base"},{"name":"const","entity":"value","module":"GHC.Base"},{"name":".","entity":"value","module":"GHC.Base"},{"name":"flip","entity":"value","module":"GHC.Base"},{"name":"$","entity":"value","module":"GHC.Base"},{"name":"until","entity":"value","module":"GHC.Base"},{"name":"asTypeOf","entity":"value","module":"GHC.Base"},{"name":"error","entity":"value","module":"GHC.Err"},{"name":"undefined","entity":"value","module":"GHC.Err"},{"name":"seq","entity":"value","module":"GHC.Prim"},{"name":"$!","entity":"value","module":"Prelude"},{"name":"map","entity":"value","module":"GHC.Base"},{"name":"++","entity":"value","module":"GHC.Base"},{"name":"filter","entity":"value","module":"GHC.List"},{"name":"head","entity":"value","module":"GHC.List"},{"name":"last","entity":"value","module":"GHC.List"},{"name":"tail","entity":"value","module":"GHC.List"},{"name":"init","entity":"value","module":"GHC.List"},{"name":"null","entity":"value","module":"GHC.List"},{"name":"length","entity":"value","module":"GHC.List"},{"name":"!!","entity":"value","module":"GHC.List"},{"name":"reverse","entity":"value","module":"GHC.List"},{"name":"foldl","entity":"value","module":"GHC.List"},{"name":"foldl1","entity":"value","module":"Data.List"},{"name":"foldr","entity":"value","module":"GHC.Base"},{"name":"foldr1","entity":"value","module":"GHC.List"},{"name":"and","entity":"value","module":"GHC.List"},{"name":"or","entity":"value","module":"GHC.List"},{"name":"any","entity":"value","module":"GHC.List"},{"name":"all","entity":"value","module":"GHC.List"},{"name":"sum","entity":"value","module":"Data.List"},{"name":"product","entity":"value","module":"Data.List"},{"name":"concat","entity":"value","module":"GHC.List"},{"name":"concatMap","entity":"value","module":"GHC.List"},{"name":"maximum","entity":"value","module":"Data.List"},{"name":"minimum","entity":"value","module":"Data.List"},{"name":"scanl","entity":"value","module":"GHC.List"},{"name":"scanl1","entity":"value","module":"GHC.List"},{"name":"scanr","entity":"value","module":"GHC.List"},{"name":"scanr1","entity":"value","module":"GHC.List"},{"name":"iterate","entity":"value","module":"GHC.List"},{"name":"repeat","entity":"value","module":"GHC.List"},{"name":"replicate","entity":"value","module":"GHC.List"},{"name":"cycle","entity":"value","module":"GHC.List"},{"name":"take","entity":"value","module":"GHC.List"},{"name":"drop","entity":"value","module":"GHC.List"},{"name":"splitAt","entity":"value","module":"GHC.List"},{"name":"takeWhile","entity":"value","module":"GHC.List"},{"name":"dropWhile","entity":"value","module":"GHC.List"},{"name":"span","entity":"value","module":"GHC.List"},{"name":"break","entity":"value","module":"GHC.List"},{"name":"elem","entity":"value","module":"GHC.List"},{"name":"notElem","entity":"value","module":"GHC.List"},{"name":"lookup","entity":"value","module":"GHC.List"},{"name":"zip","entity":"value","module":"GHC.List"},{"name":"zip3","entity":"value","module":"GHC.List"},{"name":"zipWith","entity":"value","module":"GHC.List"},{"name":"zipWith3","entity":"value","module":"GHC.List"},{"name":"unzip","entity":"value","module":"GHC.List"},{"name":"unzip3","entity":"value","module":"GHC.List"},{"name":"lines","entity":"value","module":"Data.List"},{"name":"words","entity":"value","module":"Data.List"},{"name":"unlines","entity":"value","module":"Data.List"},{"name":"unwords","entity":"value","module":"Data.List"},{"name":"ShowS","entity":"type","module":"GHC.Show"},{"name":"Show","entity":"class","module":"GHC.Show"},{"name":"showsPrec","class":"Show","entity":"method","module":"GHC.Show"},{"name":"showList","class":"Show","entity":"method","module":"GHC.Show"},{"name":"show","class":"Show","entity":"method","module":"GHC.Show"},{"name":"shows","entity":"value","module":"GHC.Show"},{"name":"showChar","entity":"value","module":"GHC.Show"},{"name":"showString","entity":"value","module":"GHC.Show"},{"name":"showParen","entity":"value","module":"GHC.Show"},{"name":"ReadS","entity":"type","module":"Text.ParserCombinators.ReadP"},{"name":"Read","entity":"class","module":"GHC.Read"},{"name":"readsPrec","class":"Read","entity":"method","module":"GHC.Read"},{"name":"readList","class":"Read","entity":"method","module":"GHC.Read"},{"name":"reads","entity":"value","module":"Text.Read"},{"name":"readParen","entity":"value","module":"GHC.Read"},{"name":"read","entity":"value","module":"Text.Read"},{"name":"lex","entity":"value","module":"GHC.Read"},{"name":"IO","entity":"newtype","module":"GHC.Types"},{"name":"putChar","entity":"value","module":"System.IO"},{"name":"putStr","entity":"value","module":"System.IO"},{"name":"putStrLn","entity":"value","module":"System.IO"},{"name":"print","entity":"value","module":"System.IO"},{"name":"getChar","entity":"value","module":"System.IO"},{"name":"getLine","entity":"value","module":"System.IO"},{"name":"getContents","entity":"value","module":"System.IO"},{"name":"interact","entity":"value","module":"System.IO"},{"name":"FilePath","entity":"type","module":"GHC.IO"},{"name":"readFile","entity":"value","module":"System.IO"},{"name":"writeFile","entity":"value","module":"System.IO"},{"name":"appendFile","entity":"value","module":"System.IO"},{"name":"readIO","entity":"value","module":"System.IO"},{"name":"readLn","entity":"value","module":"System.IO"},{"name":"IOError","entity":"type","module":"GHC.IO.Exception"},{"name":"ioError","entity":"value","module":"GHC.IO.Exception"},{"name":"userError","entity":"value","module":"GHC.IO.Exception"}]
