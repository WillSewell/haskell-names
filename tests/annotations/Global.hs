module Global where

import Prelude

foo :: Newtype
foo = Newtype foo
