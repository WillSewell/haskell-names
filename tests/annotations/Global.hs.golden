foo      at  5:1 is none
Newtype  at  5:8 is a global newtype, Prelude.Newtype
Newtype  at  5:8 is a global newtype, Prelude.Newtype
foo      at  6:1 is a value bound here
Newtype  at  6:7 is a global constructor, Prelude.Newtype
Newtype  at  6:7 is a global constructor, Prelude.Newtype
foo      at 6:15 is a global value, Global.foo
foo      at 6:15 is a global value, Global.foo
